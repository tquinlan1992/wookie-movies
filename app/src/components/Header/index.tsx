import React, { useState } from "react";
import styled from "styled-components";

const SearchInput = styled.input`
  margin-top: auto;
`;

const Column = styled.div`
  display: flex;
  flex-direction: column;
  flex-basis: 100%;
  flex: 1;
`;

const HeaderWrapper = styled.div`
  padding: 15px;
  border-bottom-style: solid;
  boder-bottom-color: black;
  border-bottom-width: 3px;
`;

const HeaderColumns = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  width: 100%;
`;

const Header: React.FC<{ onSearch: (search: string) => void }> = ({
  onSearch,
}) => {
  const [searchInputValue, setSearchInputValue] = useState("");
  const onSearchInputChange = (search: string) => {
    setSearchInputValue(search);
  };
  const performSearch: React.FormEventHandler<HTMLFormElement> = (event) => {
    event.preventDefault();
    onSearch(searchInputValue);
  };
  return (
    <HeaderWrapper>
      <HeaderColumns>
        <Column>
          <span>Wookie</span>
          <span>Movies</span>
        </Column>
        <Column>
          <form onSubmit={performSearch}>
            <SearchInput
              value={searchInputValue}
              onChange={(event) => onSearchInputChange(event.target.value)}
            />
          </form>
        </Column>
      </HeaderColumns>
    </HeaderWrapper>
  );
};

export default Header;
