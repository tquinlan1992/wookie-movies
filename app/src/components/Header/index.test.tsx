import { configure, mount } from "enzyme";
import Header from "./";
import toJson from "enzyme-to-json";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import SearchContext from "../../searchContext";
import { BrowserRouter as Router } from "react-router-dom";

configure({ adapter: new Adapter() });

test("Header", async () => {
  const onSearch = jest.fn();
  const wrapper = mount(
    <SearchContext.Provider value={{ search: "", setSearch: onSearch }}>
      <Router>
        <Header />
      </Router>
    </SearchContext.Provider>
  );
  expect(toJson(wrapper)).toMatchSnapshot();
  wrapper
    .find("input")
    .simulate("change", { target: { value: "new search value" } });
  wrapper.find("form").simulate("submit");
  expect(onSearch).toHaveBeenCalledTimes(1);
  expect(onSearch).toHaveBeenCalledWith("new search value");
});
