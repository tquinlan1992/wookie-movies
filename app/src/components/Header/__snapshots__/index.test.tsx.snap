// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Header>
      <styled.div>
        <div
          className="sc-dlfnuX kTvNJn"
        >
          <styled.div>
            <div
              className="sc-hKgJUU eSwVWB"
            >
              <styled.div>
                <div
                  className="sc-gsTEea gEuHtg"
                >
                  <Styled(Link)
                    to="/"
                  >
                    <Link
                      className="sc-eCstlR eNpSIK"
                      to="/"
                    >
                      <LinkAnchor
                        className="sc-eCstlR eNpSIK"
                        href="/"
                        navigate={[Function]}
                      >
                        <a
                          className="sc-eCstlR eNpSIK"
                          href="/"
                          onClick={[Function]}
                        >
                          <span>
                            Wookie 
                            <br />
                             Movies
                          </span>
                        </a>
                      </LinkAnchor>
                    </Link>
                  </Styled(Link)>
                </div>
              </styled.div>
              <styled.div>
                <div
                  className="sc-gsTEea gEuHtg"
                >
                  <styled.form
                    onSubmit={[Function]}
                  >
                    <form
                      className="sc-bdfBQB cmtwxi"
                      onSubmit={[Function]}
                    >
                      <input
                        onChange={[Function]}
                        value=""
                      />
                    </form>
                  </styled.form>
                </div>
              </styled.div>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </Header>
  </Router>
</BrowserRouter>
`;
